src/mpi-base.jl

```diff
@@ -373,9 +373,14 @@ function bcast(obj, root::Integer, comm::Comm)
 end
 
 function Reduce{T<:MPIDatatype}(sendbuf::Union{Ptr{T},Array{T}}, count::Integer,
-                                op::Op, root::Integer, comm::Comm)
+                                op::Op, root::Integer, comm::Comm; in_place=false)
```

Comment:
Instead of adding a new argument to `Reduce`, there should be a new function `Reduce!` that uses in-place communication.